src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const arrWithCopies = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete stateCopy[property];
+        };
+        break;
+
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => delete stateCopy[key]);
+        break;
+
+      default:
+        break;
+    }
+    arrWithCopies.push({ ...stateCopy });
```

Comment:
you have already copied it before, no need to do it here i guess
```suggestion
    arrWithCopies.push(stateCopy);
```

Reply:
If I remove here spread operator, I have errors in 2 tests that check on multiple properties (removeProperties, then addProperties, etc)